Address huf / piz performance regressions

kdt3rd opened the pull request:
On Apple OS, the fast huf decoder was accidentally disabled.

Apply some other optimizations to the huf and piz decoders

kdt3rd committed:
Enable fast huf for os/x, optimize routines

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Apply some minor optimizations to piz decode

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
disable alias tag for msvc

It appears MSVC does not handle the restrict keyword properly

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

cary-ilm reviewed (approved):
LGTM

cary-ilm commented:
@kdt3rd, does this address all of the regressions we've heard about, or is there more coming? 

I'll stage a patch release tomorrow.

kdt3rd committed:
Replicate comment denoting issue w/ msvc

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd commented:
@cary-ilm it addresses at least one issue from the TSC meeting. I have not yet performed a full regression test to know if it addresses the other performance regression that was noted, so perhaps delay a patch release until I can check that

kdt3rd committed:
Remove overzealous memory optimization

It is more efficient to store the code length and symbol in the one
table, but appears measurably slower. Remove that.

Further, do not constantly refill the bit buffer during huf fast decode,
that slows down the decoding

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd commented:
@cary-ilm this latest commit fixes the other regression, reading piz is now 15-20ms faster on 8k x 8k than the commit prior to the C-switchover on my machine (more than 150ms than 3.3.0).
